blink/memory.c

```diff
@@ -445,13 +446,31 @@ u8 *ReserveAddress(struct Machine *m, i64 v, size_t n, bool writable) {
   }
 }
 
-u8 *AccessRam(struct Machine *m, i64 v, size_t n, void *p[2], u8 *tmp,
-              bool copy) {
+#ifndef DISABLE_ROM
+bool IsRomAddress(struct Machine *m, u8 *r) {
+  if (m->metal) {
+    u8 *real = m->system->real;
+#if CAN_ABUSE_POINTERS
+    ptrdiff_t d = r - real;
+    if (kBiosOptBase <= d && d < kBiosEnd) return true;
+#else
+    if (real[kBiosOptBase] <= r && r < &real[kBiosEnd]) return true;
```

Comment:
@tkchia: I just noticed this here: I think you mean `&real[kBiosOptBase]` on the left hand side?